Binary files /home/ec2-user/guidance-for-generative-bi-v1.9.0/application/nlq/__pycache__/__init__.cpython-39.pyc and /home/ec2-user/generative-bi-using-rag/application/nlq/__pycache__/__init__.cpython-39.pyc differ
Binary files /home/ec2-user/guidance-for-generative-bi-v1.9.0/application/nlq/business/__pycache__/__init__.cpython-39.pyc and /home/ec2-user/generative-bi-using-rag/application/nlq/business/__pycache__/__init__.cpython-39.pyc differ
Binary files /home/ec2-user/guidance-for-generative-bi-v1.9.0/application/nlq/business/__pycache__/model.cpython-39.pyc and /home/ec2-user/generative-bi-using-rag/application/nlq/business/__pycache__/model.cpython-39.pyc differ
Binary files /home/ec2-user/guidance-for-generative-bi-v1.9.0/application/nlq/business/__pycache__/profile.cpython-39.pyc and /home/ec2-user/generative-bi-using-rag/application/nlq/business/__pycache__/profile.cpython-39.pyc differ
diff -ruN /home/ec2-user/guidance-for-generative-bi-v1.9.0/application/nlq/core/state_machine.py /home/ec2-user/generative-bi-using-rag/application/nlq/core/state_machine.py
--- /home/ec2-user/guidance-for-generative-bi-v1.9.0/application/nlq/core/state_machine.py	2025-02-21 09:11:33.000000000 +0000
+++ /home/ec2-user/generative-bi-using-rag/application/nlq/core/state_machine.py	2025-04-09 16:37:15.193979226 +0000
@@ -195,9 +195,18 @@
                 each_item_dict["_source"] = each_entity["_source"]
                 if "vector_field" in each_item_dict["_source"]:
                     del each_item_dict["_source"]["vector_field"]
-                entity_retrieve.append(each_item_dict)
+                if each_entity['_score'] > 0.8:
+                    entity_retrieve.append(each_item_dict)
                 if each_entity['_source']['entity_count'] > 1 and each_entity['_score'] > 0.98:
                     same_name_entity[each_entity['_source']['entity']] = each_entity['_source']['entity_table_info']
+            # 保存实体信息到answer对象中
+            self.answer.ask_entity_select.entity_retrieval = entity_retrieve
+        
+            # 如果是knowledge_search意图，直接转向KNOWLEDGE_SEARCH状态
+            if self.answer.query_intent == "knowledge_search":
+                self.transition(QueryState.KNOWLEDGE_SEARCH)
+                return
+                               
             if len(same_name_entity) > 0 and self.answer.query_intent == "normal_search":
                 for key, value in same_name_entity.items():
                     change_value = []
@@ -378,7 +387,10 @@
             self.answer.query_intent = "reject_search"
             self.transition(QueryState.REJECT_INTENT)
         elif self.knowledge_search_flag:
-            self.transition(QueryState.KNOWLEDGE_SEARCH)
+            #self.transition(QueryState.KNOWLEDGE_SEARCH)
+            self.answer.query_intent = "knowledge_search"
+            # 先进入实体检索状态
+            self.transition(QueryState.ENTITY_RETRIEVAL)
         elif self.agent_intent_flag:
             self.answer.query_intent = "agent_search"
             self.transition(QueryState.AGENT_TASK)
@@ -395,10 +407,14 @@
 
     @log_execution
     def handle_knowledge_search(self):
+        # 使用已经保存在 answer 对象中的实体信息
+        entity_info = self.answer.ask_entity_select.entity_retrieval
         response, model_response = knowledge_search(search_box=self.context.query_rewrite,
                                                     model_id=self.context.model_type,
                                                     prompt_map=self.context.database_profile["prompt_map"],
-                                                    environment_dict=self.context.database_profile["prompt_environment"])
+                                                    environment_dict=self.context.database_profile["prompt_environment"],
+                                                    entity_info=entity_info  # 添加实体信息参数
+                                                    )
         self.token_info[QueryState.KNOWLEDGE_SEARCH.name] = model_response.token_info
         self.answer.query = self.context.search_box
         self.answer.query_rewrite = self.context.query_rewrite
Binary files /home/ec2-user/guidance-for-generative-bi-v1.9.0/application/nlq/data_access/__pycache__/__init__.cpython-39.pyc and /home/ec2-user/generative-bi-using-rag/application/nlq/data_access/__pycache__/__init__.cpython-39.pyc differ
Binary files /home/ec2-user/guidance-for-generative-bi-v1.9.0/application/nlq/data_access/__pycache__/dynamo_model.cpython-39.pyc and /home/ec2-user/generative-bi-using-rag/application/nlq/data_access/__pycache__/dynamo_model.cpython-39.pyc differ
Binary files /home/ec2-user/guidance-for-generative-bi-v1.9.0/application/nlq/data_access/__pycache__/dynamo_profile.cpython-39.pyc and /home/ec2-user/generative-bi-using-rag/application/nlq/data_access/__pycache__/dynamo_profile.cpython-39.pyc differ
